lib/api/core/notifier.js

```diff
@@ -144,13 +145,56 @@ function notifyDocumentCreate (responseObject) {
  * @param {ResponseObject} responseObject object describing the document response from writeEngine
  * @return {Promise} number of notified rooms
  */
+function notifyDocumentReplace (responseObject) {
+  var
+    matchedRooms,
+    deferred = q.defer();
+
+  responseObject.scope = 'in';
+
+  this.dsl.testFilters(responseObject)
```

Comment:
We already pushed the result of `dsl.testFilters` in the `publish` method, to avoid calling it twice, as it is an expensive method performance-wise.
So this line must be replaced by a cache retrieval, like it is done in the `notifyDocumentCreate` method.